packages/context/test/unit/context.unit.ts

```diff
@@ -681,7 +694,223 @@ describe('Context', () => {
+      expect(ctx.isSubscribed(nonMatchingObserver)).to.false();
+    });
+
+    function givenNonMatchingObserver() {
+      nonMatchingObserver = {
+        filter: binding => false,
+        observe: (event, binding) => {},
+      };
+    }
+  });
+
+  describe('event notification', () => {
+    const events: string[] = [];
+    let nonMatchingObserverCalled = false;
+
+    beforeEach(givenObservers);
+
+    it('emits one bind event to matching observers', async () => {
+      ctx.bind('foo').to('foo-value');
+      await ctx.waitUntilObserversNotified();
+      expect(events).to.eql(['1:foo:foo-value:bind', '2:foo:foo-value:bind']);
+      expect(nonMatchingObserverCalled).to.be.false();
+    });
+
+    it('emits multiple bind events to matching observers', async () => {
+      ctx.bind('foo').to('foo-value');
+      ctx.bind('xyz').to('xyz-value');
+      await ctx.waitUntilObserversNotified();
+      expect(events).to.eql([
+        '1:foo:foo-value:bind',
+        '2:foo:foo-value:bind',
+        '1:xyz:xyz-value:bind',
+        '2:xyz:xyz-value:bind',
+      ]);
+    });
+
+    it('emits unbind event to matching observers', async () => {
+      ctx.bind('foo').to('foo-value');
+      await ctx.waitUntilObserversNotified();
+      ctx.unbind('foo');
+      await ctx.waitUntilObserversNotified();
+      expect(events).to.eql([
+        '1:foo:foo-value:bind',
+        '2:foo:foo-value:bind',
+        '1:foo:foo-value:unbind',
+        '2:foo:foo-value:unbind',
+      ]);
+      expect(nonMatchingObserverCalled).to.be.false();
+    });
+
+    it('does not trigger observers if affected binding is the same', async () => {
+      const binding = ctx.bind('foo').to('foo-value');
+      await ctx.waitUntilObserversNotified();
+      expect(events).to.eql(['1:foo:foo-value:bind', '2:foo:foo-value:bind']);
+      ctx.add(binding);
+      await ctx.waitUntilObserversNotified();
+      expect(events).to.eql(['1:foo:foo-value:bind', '2:foo:foo-value:bind']);
+    });
+
+    it('reports error if an observer fails', () => {
+      ctx.bind('bar').to('bar-value');
+      return expect(ctx.waitUntilObserversNotified()).to.be.rejectedWith(
+        'something wrong',
+      );
+    });
+
+    function givenObservers() {
+      nonMatchingObserverCalled = false;
+      events.splice(0, events.length);
+      // An observer does not match the criteria
+      const nonMatchingObserver: ContextObserver = {
+        filter: binding => false,
+        observe: (event, binding) => {
+          nonMatchingObserverCalled = true;
+        },
+      };
+      // A sync observer matches the criteria
+      const matchingObserver: ContextObserver = {
+        observe: (event, binding, context) => {
+          // Make sure the binding is configured with value
+          // when the observer is notified
+          const val = binding.getValue(context);
+          events.push(`1:${binding.key}:${val}:${event}`);
```

Comment:
I find the prefixes `1` and `2` as too cryptic. Can you use `SYNC` and `ASYNC` please?

Reply:
fixed